app/services/doc_auth/socure/request.rb

```diff
@@ -6,7 +6,7 @@ class Request
       def fetch
         # return DocAuth::Response with DocAuth::Error if workflow is invalid
         http_response = send_http_request
-        if http_response&.success? && http_response.body.present?
+        if http_response&.success? && http_response.body.force_encoding('ASCII-8BIT').present?
```

Comment:
is this `force_encoding` safe to do on all these requests, or should i pull out a method that i can overwrite in my ImagesRequest class?

Reply:
when the response body is a zip archive, `http_response.body.present?` leads to a `invalid byte sequence in UTF-8` ArgumentError :(

Reply:
Hm, would `http_response.body && http_response.body.length > 0` suffice?

Reply:
specs pass, so it suffices!